src/chain/chain.rs

```diff
@@ -1800,6 +1828,16 @@ impl EldersChangeBuilder {
     }
 }
 
+#[derive(Error, Debug)]
+pub enum ProofError {
```

Comment:
I was thinking about it, but it seems to me that when you are about to create an enum where one of the variant is basically `Ok`, then it might be more idiomatic to use `Result` instead. But not 100% convinced about it.

> It seem we are trying to reduce the number of error,

I'm not convinced that reducing the number of error overall is a meaningful goal. I think having these internal small, locally-scoped errors can make the code actually simpler to follow.

Reply:
> I think having these internal small, locally-scoped errors can make the code actually simpler to follow.

Could be, not sure, I'm not 100% convinced by that example that it makes code simpler to follow.